### PR TITLE
[dv/testplan] update description for jtag/sram execution test.

### PR DESCRIPTION
**Pull request by tjaychen**
- clarify the need to execute this test from mask rom
- also add details on what should be done to test correct
  behavior.

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
@timothytrippel let me know if this makes sense to you. I also wasn't sure what tag I was supposed to use to delineate a mask rom test, so let me know.

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 1]
```diff
@@ -529,15 +529,23 @@
...
+              main SRAM at a fixed location (bottom of SRAM).
+            - Execute a "wait_for_interrupt" in a wrapping function in the test, this causes Ibex
+              to populate the return address and halt execution.
+            - Trigger a debug request by programming `dmcontrol.haltreq` via JTAG.
+            - Program the program buffer via JTAG with code to jump Ibex to the SRAM return
+              instruction.
+            - Verfy that Ibex can fetch and execute the instructions from the SRAM correctly, Ibex
+              should return to the code that originally invoked the wait_for_interrupt routine.
+            - Release Ibex from halt via JTAG and allow the test to complete.
+            - The test should also ensure that no other interrupts occurs in the system.
```
Comment (timothytrippel) [inline hw/top_earlgrey/data/chip_testplan.hjson:545]:
```suggestion
            - The test should also ensure that no other interrupts occur in the system.
```

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 2]
```diff
@@ -529,15 +529,23 @@
...
+            access and must be used to ensure the epmp configurations are done correctly.
+
+            - X-ref'ed with chip_sw_sram_execution and populate a single "return" instruction into
+              main SRAM at a fixed location (bottom of SRAM).
+            - Execute a "wait_for_interrupt" in a wrapping function in the test, this causes Ibex
+              to populate the return address and halt execution.
+            - Trigger a debug request by programming `dmcontrol.haltreq` via JTAG.
+            - Program the program buffer via JTAG with code to jump Ibex to the SRAM return
+              instruction.
+            - Verfy that Ibex can fetch and execute the instructions from the SRAM correctly, Ibex
```
Comment (timothytrippel) [inline hw/top_earlgrey/data/chip_testplan.hjson:542]:
```suggestion
            - Verify that Ibex can fetch and execute the instructions from the SRAM correctly, Ibex
```

File: hw/top_earlgrey/data/chip_testplan.hjson [hunk 3]
```diff
@@ -529,15 +529,23 @@
...
-              via JTAG.
-            - Ibex steps into interactive debug to execute instructions in SRAM.
-            - Load instruction data into the SRAM via SW. X-ref'ed with chip_sw_sram_execution.
-            - Program the program buffer via JTAG with code to let Ibex jump to the SRAM to
-              execute the instructions.
-            - Verfy that Ibex can fetch and execute the instructions from the SRAM correctly.
-            '''
-      milestone: V2
+            This test must be run with mask ROM.  Mask ROM is responsible for enabling debug ROM
+            access and must be used to ensure the epmp configurations are done correctly.
```
Comment (timothytrippel) [inline hw/top_earlgrey/data/chip_testplan.hjson:533]: I think we probably want to move this test-point to the `mask_rom_testplan.hjson` [here](https://github.com/lowRISC/opentitan/blob/master/sw/device/silicon_creator/mask_rom/data/mask_rom_testplan.hjson) (which gets linked to this testplan at the top via the include statements).

**Review by timothytrippel — APPROVED** [review]
I think we just need to move this to the `sw/device/silicon_creator/mask_rom/data/mask_rom_testplan.hjson` file but otherwise the description LGTM. thanks @tjaychen !

**Review by msfschaffner — APPROVED** [review]
LGTM, module @timothytrippel's suggestion.

**Comment by tjaychen** [comment]
> I think we just need to move this to the `sw/device/silicon_creator/mask_rom/data/mask_rom_testplan.hjson` file but otherwise the description LGTM. thanks @tjaychen !

ah! got it!

**Comment by tjaychen** [comment]
@timothytrippel 
@cindychip 
@weicaiyang 

is this okay? I wanted to leave a note for it in the normal testplan to make sure we don't forget. This really is more of a DV test, it just piggybacks on a few things mask_rom does.

**Review by timothytrippel — APPROVED** [review]
Yep LGTM!

**Review by msfschaffner — APPROVED** [review]

**Pull request closed without merging**